internal/pkg/cli/task_run.go

```diff
@@ -259,16 +261,54 @@ func (o *runTaskOpts) Validate() error {
 	return nil
 }
 
+func (o *runTaskOpts) validateRunnerTypeOptions() error {
+	// Subnets cannot be specified with default.
+	if o.defaultEnv && o.subnets != nil {
+		return fmt.Errorf("cannot specify both subnets and default")
+	}
+
+	// Application and environment cannot be specified with default.
+	if o.appName != "" && o.defaultEnv {
+		return fmt.Errorf("cannot specify both application and default")
+	}
+
+	if o.env != "" && o.defaultEnv {
+		return fmt.Errorf("cannot specify both environment and default")
+	}
+
+	// Application and environment cannot be specified with subnets or security groups.
+	if o.appName != "" && o.subnets != nil {
+		return fmt.Errorf("cannot specify both subnets and application")
+	}
+
+	if o.appName != "" && o.securityGroups != nil {
+		return fmt.Errorf("cannot specify both security groups and application")
+	}
+
+	if o.env != "" && o.subnets != nil {
+		return fmt.Errorf("cannot specify both subnets and environment")
+	}
+
+	if o.env != "" && o.securityGroups != nil {
+		return fmt.Errorf("cannot specify both security groups and environment")
+	}
+
+	return nil
+}
+
 // Ask prompts the user for any required or important fields that are not provided.
 func (o *runTaskOpts) Ask() error {
 	if err := o.askTaskGroupName(); err != nil {
 		return err
 	}
-	if err := o.askAppName(); err != nil {
-		return err
-	}
-	if err := o.askEnvName(); err != nil {
-		return err
+
+	if !o.defaultEnv && o.subnets == nil {
```

Comment:
This logic is so complex! Are there any other possibilities here? For instance is it okay to specify security groups and subnets, or both security groups and default? Let's make sure we're hitting all these test cases.

Reply:
I agree that it is very complex :( I have added more test cases and modify the logic a little bit. Hopefully it is clearer now.

It is okay to specify --security-groups and --subnets at the same time. It is also fine to specify both --sg and --default at the same time. 

In short, --subnets, --default, and --app/--env are mutually exclusive. --sg is compatible with --subnets, or --default, but is not compatible with --app/--env.